### PR TITLE
Wrap build_sysimg.jl script in a module

### PR DESCRIPTION
**Pull request by musm**
Wrap build_sysimg.jl script in a module, ref https://github.com/JuliaLang/julia/issues/21234; thoughts?

### CODE REVIEW

File: contrib/build_sysimg.jl [hunk 1]
```diff
@@ -172,7 +180,7 @@ end
 #
 # Also check whether we are running `genstdlib.jl`, in which case we don't want to build a
 # system image and instead only need `build_sysimg`'s docstring to be available.
-if !isdefined(Main, :GenStdLib) && !isinteractive()
```
Comment (musm) [inline contrib/build_sysimg.jl]: I think this can now be removed?

**Comment by tkelman** [comment]
Other than fixing the documentation discrepancy, what does this accomplish? Existing documentation examples about using this script would need to be updated with either `using BuildSysImg` or calling the relevant function as `BuildSysImg.build_sysimg`

**Comment by musm** [comment]
I was hoping to clean up the discrepancy between the docs, which are not correct and the script. I don't see any negatives to wrapping the script in a module tbh, seems to me cleaner, esp if needed in other codes e.g. https://github.com/dhoegh/BuildExecutable.jl/blob/master/src/BuildExecutable.jl

**Comment by musm** [comment]
ops accidently slipped on reopen when trying to close

**Pull request closed without merging**